Sources/OpenAPIRuntime/URICoder/Parsing/URIParser.swift

```diff
@@ -205,6 +213,45 @@ extension URIParser {
             }
         }
     }
+    
+    /// Parses the root node assuming the raw string uses the deepObject style
+    /// and the explode parameter is enabled.
+    /// - Returns: The parsed root node.
+    /// - Throws: An error if parsing fails.
+    private mutating func parseExplodedDeepObjectRoot() throws -> URIParsedNode {
+        try parseGenericRoot { data, appendPair in
+            let keyValueSeparator: Character = "="
+            let pairSeparator: Character = "&"
+            let nestedKeyStartingCharacter: Character = "["
+            let nestedKeyEndingCharacter: Character = "]"
+            
+            func nestedKey(from deepObjectKey: String.SubSequence) -> Raw {
+                var unescapedFirstValue = Substring(deepObjectKey.removingPercentEncoding ?? "")
```

Comment:
Here you should verify that the first character is `[`, otherwise this is a malformed key. For the rest of the characters, you should be able to use `parseUpToCharacterOrEnd` as originally was.